### PR TITLE
Testing: migrate reporting tests from PHPUnit to Cypress 

### PR DESCRIPTION
**Pull request by ifarzana**
- Relates to https://github.com/xibosignageltd/xibo-private/issues/220

### CODE REVIEW

File: cypress/e2e/report_distribution.cy.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        cy.contains('Next').should('be.visible').click();
+
+        // Should have media stats
+        cy.get('#distributionTbl tbody tr:nth-child(3) td:nth-child(1)').contains('12:00 PM'); // Period
+        cy.get('#distributionTbl tbody tr:nth-child(3) td:nth-child(2)').contains(60); // Duration
+        cy.get('#distributionTbl tbody tr:nth-child(3) td:nth-child(3)').contains(1); // Count
+
+        // ------
+        // ------
+        // Create a Daily Distribution Report Schedule
```
Comment (maurofmferrao) [inline cypress/e2e/report_distribution.cy.js:73]: Shouldn't this be a separate test?

File: cypress/e2e/report_proofofplay.cy.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        cy.visit('/report/form/proofofplayReport');
+
+        // Click on the Apply button
+        cy.contains('Apply').click();
+
+        cy.get('#stats').should('be.visible');
+
+        // Should have layout stat
+        cy.get('#stats tbody tr:nth-child(1) td:nth-child(1)').contains('layout'); // stat type
```
Comment (maurofmferrao) [inline cypress/e2e/report_proofofplay.cy.js:96]: In all the tests in this page, the UX flow is the same, only the element testing is different. To increase speed, shouldn't all tests be in a single test that tested all elements?

File: cypress/e2e/report_summary.cy.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        cy.contains('Next').should('be.visible').click();
+
+        // Should have media stats
+        cy.get('#summaryTbl tbody tr:nth-child(3) td:nth-child(1)').contains('12:00 PM'); // Period
+        cy.get('#summaryTbl tbody tr:nth-child(3) td:nth-child(2)').contains(60); // Duration
+        cy.get('#summaryTbl tbody tr:nth-child(3) td:nth-child(3)').contains(1); // Count
+
+        // ------
+        // ------
+        // Create a Daily Summary Report Schedule
```
Comment (maurofmferrao) [inline cypress/e2e/report_summary.cy.js:72]: Like the one above, shouldn't this one be a separate test?

**Review by maurofmferrao — CHANGES_REQUESTED** [review]
I just have some minor remarks ( mostly in structure of the tests ). Everything else looks good!

File: cypress/e2e/report_distribution.cy.js (hunk 1 above)
Comment (ifarzana) [reply cypress/e2e/report_distribution.cy.js:73]: I did not separate it, because there are several steps between line 30 and 55 that must be carried out.

Comment (maurofmferrao) [reply cypress/e2e/report_distribution.cy.js:73]: So maybe you can update the description of the test on line 28 just so we can know what both tests are?

**Review by maurofmferrao — APPROVED** [review]
Looks good to me

**Pull request merged**